dev-lua/sol2/sol2-3.2.2.ebuild

```diff
@@ -0,0 +1,86 @@
+# Copyright 2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+LUA_COMPAT=( lua5-{1..4} luajit )
+inherit cmake lua
+
+DESCRIPTION="C++ <-> Lua API wrapper"
+HOMEPAGE="https://github.com/ThePhD/sol2"
+SRC_URI="https://github.com/ThePhD/sol2/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+IUSE="test"
+REQUIRED_USE="${LUA_REQUIRED_USE}"
+RESTRICT="!test? ( test )"
+
+RDEPEND="${LUA_DEPS}"
+DEPEND="
+	test? (
+		dev-cpp/catch:1
+		${LUA_DEPS}
+	)
+"
+BDEPEND="
+	test? (
+		virtual/pkgconfig
+	)
+"
+
+PATCHES=(
+	"${FILESDIR}"/${P}-luajit-pkgconf.patch
+	"${FILESDIR}"/${P}-catch-depend.patch
+)
+
+src_prepare() {
+	sed -i -e 's/-Werror//' \
+		tests/runtime_tests/CMakeLists.txt \
+		tests/regression_tests/1011/CMakeLists.txt \
+		tests/config_tests/function_pointers/CMakeLists.txt \
+		examples/customization/CMakeLists.txt \
+		examples/CMakeLists.txt || die
+	grep -rn Werror && die "Werror is not removed from somewhere"
```

Comment:
That's some nasty Werror proliferation.

If it's in that many files and it's safe to sed them all, may be worth just using:
`find . -name CMakeLists.txt -exec sed -i 's/-Werror//' {} + || die`
It won't miss new ones either so the grep won't be particularly useful.

Downside of find is that it won't tell us if no files were found, but I'd like to think we'll notice if it migrated away from cmake.

Reply:
They already removed all these Werrors since the release. So for the next release this sed should become much better (removed also, together with grep)

Reply:
When have that knowledge, it's nice to leave note in the ebuild in case of an event where someone else did the bump.

Reply:
I actually tried to package the latest git snapshot :)
They also renamed the CMakeLists.txt directories, so this sed line is failing there already. That's hard to miss while doing the bump.

Reply:
Why keep the grep in this current ebuild then, the only way it made sense to me is "detect changes on bump".

Reply:
True. Fixed.